### PR TITLE
Adds support of DATM_MODE=CLM1PT for ELM

### PR DESCRIPTION
**Pull request by bishtgautam**
Modifications to support user-defined atmospheric forcing
with ELM.

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by bishtgautam** [comment]
The new code in [src/components/data_comps_mct/datm/cime_config/buildnml ](https://github.com/ESMCI/cime/pull/3845/files#diff-d404eb6de5990e8b29112276e0189e7a1635a024660d96e52f138e78d9e4dfe5R47) may not be an ideal solution, but I wanted to open this PR to get the conversation started on how to best accommodate these changes.

**Review by jedwards4b — APPROVED** [review]
I don't see any problems with this.

**Comment by bishtgautam** [comment]
It seems I broke a test.

**Comment by jedwards4b** [comment]
You did - do you know where to find the log?   I also notice that you didn't run scripts_regression_tests.py locally for this PR.   Can you please do that and update the initial comment of the conversation.

```
======================================================================
FAIL: test_pylint_src_components_data_comps_mct_datm_cime_config_buildnml (__main__.B_CheckCode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 3480, in test
    self.assertTrue(result == "", msg=result)
AssertionError: False is not true : ************* Module buildnml
W: 57, 0: Bad indentation. Found 13 spaces, expected 12 (bad-indentation)
W: 59, 0: Bad indentation. Found 13 spaces, expected 12 (bad-indentation)
W: 65, 0: Bad indentation. Found 13 spaces, expected 12 (bad-indentation)
W: 67, 0: Bad indentation. Found 13 spaces, expected 12 (bad-indentation)
```

File: src/components/data_comps_mct/datm/cime_config/buildnml [hunk 1]
```diff
@@ -44,18 +44,30 @@ def _create_namelists(case, confdir, inst_string, infile, nmlgen, data_list_path
...
     # Check for incompatible options.
     #----------------------------------------------------
-    if "CLM" in datm_mode and case.get_value("COMP_LND") == "clm":
-        expect(datm_presaero != "none",
-               "A DATM_MODE for CLM is incompatible with DATM_PRESAERO=none.")
-        expect(datm_topo != "none",
-               "A DATM_MODE for CLM is incompatible with DATM_TOPO=none.")
-    expect(grid != "CLM_USRDAT" or clm_usrdat_name in ("", "UNSET"),
-           "GRID=CLM_USRDAT and CLM_USRDAT_NAME is NOT set.")
+    if get_model() != "e3sm":
```
Comment (jgfouca) [inline src/components/data_comps_mct/datm/cime_config/buildnml:55]: The code looks identical in both branches with the only difference being ELM vs CLM. You should capture this difference in a single string so the code can be reused.

**Review by jgfouca — CHANGES_REQUESTED** [review]

**Review by jedwards4b — APPROVED** [review]
Better, thanks.

**Comment by rljacob** [comment]
@jgfouca this should be merged if you're happy with it.

**Review by jgfouca — APPROVED** [review]

**Pull request merged**